assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidgetGA4.stories.js

```diff
@@ -0,0 +1,397 @@
+		endDate: '2020-09-07',
+		dimensions: [ 'date' ],
+		metrics: [
+			{
+				name: 'screenPageViews',
+			},
+			{
+				name: 'sessions',
+			},
+			{
+				name: 'engagedSessions',
+			},
+			{
+				name: 'averageSessionDuration',
+			},
+		],
+		orderby: [
+			{
+				dimension: {
+					dimensionName: 'date',
+				},
+			},
+		],
+		url: null,
+	},
+	{
+		dimensions: [ 'date' ],
+		metrics: [ { name: 'totalUsers' } ],
+		startDate: '2020-08-11',
+		endDate: '2020-09-07',
+	},
+];
+const currentEntityURL = 'https://www.example.com/example-page/';
+const reportOptionsWithEntity = reportOptions.map( ( options ) => {
+	return {
+		...options,
+		url: currentEntityURL,
+	};
+} );
+
+const WidgetWithComponentProps = withWidgetComponentProps( 'widget-slug' )(
+	DashboardOverallPageMetricsWidgetGA4
+);
+
+const Template = ( { setupRegistry, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry }>
+		<WidgetWithComponentProps { ...args } />
+	</WithRegistrySetup>
+);
+
+export const Ready = Template.bind( {} );
+Ready.storyName = 'Ready';
+Ready.args = {
+	setupRegistry: ( registry ) => {
+		for ( const options of reportOptions ) {
+			provideAnalytics4MockReport( registry, options );
+		}
+	},
+};
+Ready.scenario = {
+	label: 'Modules/Analytics/Widgets/DashboardOverallPageMetricsWidgetGA4/Ready',
+};
+
+export const Loading = Template.bind( {} );
+Loading.storyName = 'Loading';
+Loading.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( MODULES_ANALYTICS_4 ).startResolution( 'getReport', [
+			reportOptions[ 0 ],
+		] );
+	},
+};
+Loading.scenario = {
+	label: 'Modules/Analytics/Widgets/DashboardOverallPageMetricsWidgetGA4/Loading',
+};
+
+export const DataUnavailable = Template.bind( {} );
+DataUnavailable.storyName = 'Data Unavailable';
+DataUnavailable.args = {
+	setupRegistry: ( { dispatch } ) => {
+		const propertyID = properties[ 0 ]._id;
+		// Set the property creation timestamp to one and a half days ago, so that
+		// the property is considered to be in the gathering data state.
+		const createTime = new Date(
+			Date.now() - DAY_IN_SECONDS * 1.5 * 1000
+		).toISOString();
+
+		const property = {
+			...properties[ 0 ],
+			createTime,
+		};
+
+		dispatch( MODULES_ANALYTICS_4 ).receiveGetProperty( property, {
+			propertyID,
+		} );
+		dispatch( MODULES_ANALYTICS_4 ).setPropertyID( propertyID );
+
+		dispatch( MODULES_ANALYTICS_4 ).receiveGetReport(
+			replaceValuesInAnalytics4ReportWithZeroData(
+				getAnalytics4MockResponse( gatheringReportOptions )
+			),
+			{
+				options: gatheringReportOptions,
+			}
+		);
+
+		for ( const options of reportOptions ) {
+			dispatch( MODULES_ANALYTICS_4 ).receiveGetReport(
+				replaceValuesInAnalytics4ReportWithZeroData(
+					getAnalytics4MockResponse( options )
+				),
+				{ options }
+			);
+		}
+	},
+};
+DataUnavailable.scenario = {
+	label: 'Modules/Analytics/Widgets/DashboardOverallPageMetricsWidgetGA4/DataUnavailable',
+};
+
+export const ZeroData = Template.bind( {} );
```

Comment:
Note, there are no sparklines for the Zero Data story - this will be addressed as necessary in #6623.